streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -58,19 +60,18 @@
     private boolean transactionInitialized = false;
 
     public StreamsProducer(final Producer<byte[], byte[]> producer,
-                           final boolean eosEnabled,
-                           final LogContext logContext,
-                           final String applicationId) {
-        log = logContext.logger(getClass());
+                           final String applicationId,
+                           final LogContext logContext) {
+        log = Objects.requireNonNull(logContext.logger(getClass()), "logContext cannot be null");
         logPrefix = logContext.logPrefix().trim();
 
         this.producer = Objects.requireNonNull(producer, "producer cannot be null");
         this.applicationId = applicationId;
-        this.eosEnabled = eosEnabled;
+        this.eosEnabled = applicationId != null;
     }
 
     private String formatException(final String message) {
-        return message + " [" + logPrefix + ", " + (eosEnabled ? "eos" : "alo") + "]";
+        return message + " [" + logPrefix + "]";
```

Comment:
It's redundant to log if EOS is enabled or not; it's a single global config. Not need to include.

Reply:
I would rather we keep it here so that we don't need to look around the log to identify the latest restart log to determine whether we are on EOS or not.

Reply:
If we argue this way, you could also justify to log the whole config each time :)